src/k8s/pkg/k8sd/types/cluster_config_datastore.go

```diff
@@ -24,3 +29,45 @@ func (c Datastore) GetExternalClientKey() string  { return getField(c.ExternalCl
 func (c Datastore) Empty() bool {
 	return c.Type == nil && c.K8sDqlitePort == nil && c.K8sDqliteCert == nil && c.K8sDqliteKey == nil && c.ExternalURL == nil && c.ExternalCACert == nil && c.ExternalClientCert == nil && c.ExternalClientKey == nil
 }
+
+// DatastoreSnapProvider is to avoid circular dependency for snap.Snap in Datastore.ToKubeAPIServerArguments()
+type DatastorePathsProvider interface {
+	K8sDqliteStateDir() string
+	EtcdPKIDir() string
+}
+
+// ToKubeAPIServerArguments returns updateArgs, deleteArgs that can be used with snaputil.UpdateServiceArguments() for the kube-apiserver
+// according the datastore configuration.
+func (c Datastore) ToKubeAPIServerArguments(p DatastorePathsProvider) (map[string]string, []string) {
+	var (
+		updateArgs = make(map[string]string)
+		deleteArgs []string
+	)
+
+	switch c.GetType() {
+	case "k8s-dqlite":
+		updateArgs["--etcd-servers"] = fmt.Sprintf("unix://%s", path.Join(p.K8sDqliteStateDir(), "k8s-dqlite.sock"))
+		deleteArgs = []string{"--etcd-cafile", "--etcd-certfile", "--etcd-keyfile"}
+	case "external":
+		updateArgs["--etcd-servers"] = c.GetExternalURL()
```

Comment:
we should keep in mind that this is really `ExternalURLs` -- `etcd-servers` is a comma separated list of  servers.  I sort of wish the data structure was a `[]string` rather than `string`.

> List of etcd servers to connect with (scheme://ip:port), comma separated.

The endpoints can be https or http of course, so the cert fields are only necessary for `https`

Reply:
The API can be `[]string`, regardless of the internal type.